### ISSUE TITLE
[Help] Local data lost on switching from DEB to Snap install

### ISSUE THREAD

**Issue by owais** [getinsomnia/insomnia#1614]
I had an old Insomnia copy installed from a DEB file on Ubuntu 19.04. I decided to switch to snaps. Uninstalled the deb version and installed one from the snap store. Next time I fired up insomnia, I couldn't see any of my  request definitions. I'm not sure if the switch to snaps caused it or if the new version was somehow incompatible with the old one. I don't know what the old version was. 

The one I installed from snaps is ` 6.5.4`. 

P.S, I also tried to find the local insomnia files but didn't see them in home, .cache, .local/share or .config. Is there any chance I can recover the older files, install an old version, export the data and then import it in the new version?

**Comment by gschier** [comment]
Hey @owais,

The data for the deb install lives in `~/.config/Insomnia` I'm pretty sure. You can find the data directory from Help > Show App Data Folder. I have a feeling the Snap data is stored somewhere else and it's just not finding it.

For reference, here's what Electron states as the app data folders for each platform:

- `%APPDATA%/Insomnia` on Windows
- `$XDG_CONFIG_HOME/Insomnia` or `~/.config/Insomnia` on Linux
- `~/Library/Application Support/Insomnia` on macOS

**Comment by owais** [comment]
Hi @gschier. I just figured it out. I missed the obvious fact that snaps are confined and cannot access any directory other than the one snap provides them. I installed the unconfined DEB version, exported the workspaces from there and imported into the snap version.

**Comment by amcsi** [comment]
I got all my Insomnia data wiped using Snap after an upgrade of Insomnia.

Luckily it was simple to recover my data.

To anyone who may run into this in the future:

Go to `~/snap/insomnia`.
See that `current` is a link to the latest version of the app.
Move the latest version folder to a backup.
Copy the version folder that's one version older recursively in place of the latest version's folder.

By doing this I was able to get back all my Insomnia data.
